src/material/schematics/ng-generate/m3-theme/README.md

```diff
@@ -15,33 +15,80 @@ secondary, tertiary, and neutral colors. It is recommended to choose colors that
-mixins.
-
-If you're using the system variables option, you should remember to either provide values for the
-system variables (all prefixed with `--sys-`), or to include the `system-level-colors` and
-`system-level-typography` mixins which will generate the values based on your theme.
-
-The default prefix for system variables is `--sys-`. This prefix can be customized. For
-example, to change the prefix to `--md-sys-`, use the following configuration at the color or typography level:
-`system-prefix: md-sys`.
+specified directory or the project root with the generated palettes. The exported
+palettes (`$primary-palette` and `$tertiary-palette`) can be provided to the `private-experimental-theme` mixin within your theme file to use the custom colors.
 
 ```scss
 @use '@angular/material' as mat;
-@use './path/to/my-theme';
+@use './path/to/my-theme'; // location of generated file
+
+html {
+  @include mat.private-experimental-theme(
+    color: (
+      primary: my-theme.$primary-palette,
+      tertiary: my-theme.$tertiary-palette,
+    ),
+  )
+}
+```
 
-@include mat.core();
+High contrast override theme mixins are also generated in the file if specified
+(`high-contrast-light-theme-overrides` and `high-contrast-dark-theme-overrides`). These mixins
+override the system level variables with high contrast equivalent values from your theme. This is
+helpful for users who prefer more contrastful colors for either preference or accessibility reasons.
+
+```scss
+@use '@angular/material';
+@use './path/to/my-theme'; // location of generated file
 
 html {
   // Apply the light theme by default
-  @include mat.core-theme(my-theme.$light-theme);
-  @include mat.button-theme(my-theme.$light-theme);
+  @include material.private-experimental-theme((
+    color: (
+      primary: my-theme.$primary-palette,
+      tertiary: my-theme.$tertiary-palette,
+    ),
+  ));
+
+  // Use high contrast light theme colors when users prefer contrast
+  @media (prefers-contrast: more) {
+    @include my-theme.high-contrast-light-theme-overrides();
+  }
+
+  // Apply dark theme when users prefer a dark color scheme
+  @media (prefers-color-scheme: dark) {
+    @include material.private-experimental-theme((
+      color: (
+        primary: my-theme.$primary-palette,
+        tertiary: my-theme.$tertiary-palette,
+        theme-type: dark,
+      ),
+    ));
+
+    // Use high contrast dark theme colors when users prefers a dark color scheme and contrast
+    @media (prefers-contrast: more) {
+      @include my-theme.high-contrast-dark-theme-overrides();
+    }
+  }
+}
+```
 
-  // When using system variables, remember to provide values for them
-  // or uncomment the lines below to generate them from the theme.
-  // @include mat.system-level-colors(my-theme.$light-theme);
-  // @include mat.system-level-typography(my-theme.$light-theme);
+The `private-experimental-theme` mixin will become the default and recommended way of defining
```

Comment:
I think we can skip this section and assume users are going to be guided towards `theme` only. For users with `define-theme` already, I'd expect them to already understand using palettes